### ISSUE TITLE
Keras load weights does not work

### ISSUE THREAD

**Issue by ybsave** [tensorflow/tensorflow#21460]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Y
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0/7
- **GPU model and memory**: Quardo M4000
- **Exact command to reproduce**: 

### Describe the problem
I try a simple Keras model. After saving the model weights, I tried to reload it. However, the "load_weights" function does not really load the weights, but still use the initializer's setting in the model.

### Source code / logs
	def test_load():
	  data = np.ones((1,1,4,5), dtype=float)
	  
	  model = keras.Sequential()
	  model.add(Conv2D(5, 1, use_bias=False, kernel_initializer='ones', trainable=False))

	  results = model.predict(data)
	  print(results)

	  for layer in model.layers:
	    weights = layer.get_weights()
	    layer.set_weights(np.zeros_like(weights))

	  results = model.predict(data)
	  print(results)

	  model.save_weights('./temp')

	  new_model = keras.Sequential()
	  new_model.add(Conv2D(5, 1, use_bias=False, kernel_initializer='uniform', trainable=False))
	  
	  new_model.load_weights('./temp')

	  results = new_model.predict(data)
	  print(results)

Before saving the model, everything works as expected. However, after loading, the loaded weights are always random according to the initializer. I wonder whether there is some bugs in the codes or I used it in a wrong way. Thank you.

**Comment by allenlavoie** [comment]
Sorry you ran into this. There was a bug with when restore ops got run when graph building; the correct behavior was to mimic eager execution (run restore ops in the Session as they came in), but they were getting run just once instead (which probably interacts with Sequential building immediately here).

Your snippet works for me in the 2018-06-28 nightly. The fix was https://github.com/tensorflow/tensorflow/commit/21d94be838f7a7917f072e5ec0bbe6e3593177b9#diff-7b3f8757e2c5c405c3d056aa0a301c8f

Could you give a nightly a try?

**Comment by allenlavoie** [comment]
You're right, 20180802 was the one that worked. 20180628 was broken. I don't expect a regression, so anything from August should work.

Looks like the only August Windows builds are CPU: https://pypi.org/project/tf-nightly/1.10.0.dev20180802/#files

**Comment by ybsave** [comment]
Thank you for the clarification. Do you know whether there be any Windows builds for tensorflow GPU version soon?

**Comment by allenlavoie** [comment]
@av8ramit are we planning to do Windows GPU nightly builds?

**Comment by av8ramit** [comment]
@allenlavoie we are in the process of migrating to bazel from cmake. Hopefully that fixes that build.

**Comment by ybsave** [comment]
@av8ramit Thank you for your update. Once the new build is available, I will test it and report the results here.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @allenlavoie: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by av8ramit** [comment]
We now have Windows tf-nightly-gpu builds.

**Comment by ybsave** [comment]
I tested the new nightly built version, and the program now works well. But the Tensorflow 1.10 version does not work.

**Comment by ybsave** [comment]
Hello @allenlavoie, sorry to bother you again. Would you please help have a look at https://github.com/tensorflow/tensorflow/issues/22062? Could this be another similar bug in Keras save/load? Or it was just my mis-use? Thank you.
